geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreStatsTest.scala

```diff
@@ -389,6 +392,37 @@ class AccumuloDataStoreStatsTest extends Specification with TestWithMultipleSfts
         ds.stats.getMinMax[Int](sft, "age").map(_.tuple) must beSome((1, 2, 2L))
         ds.stats.getMinMax[Int](sft, "height") must beNone
       }
+
+      "calculate exact counts with index query hints" >> {
+        // mock accumulo seems to really struggle with setting up the stat iterator
+        // the call to batchScanner.iterator can take several seconds
+        // this doesn't seem to happen with regular accumulo
+        // so only test one of each query type here
+
+        // deleting the gm.AccumuloDataStoreStatsTest_AccumuloDataStoreStatsTest0_join_name_geom_dtg_v8 table
+        // to show that the QUERY_INDEX hint is being passed through
+        ds.connector.tableOperations().delete("gm.AccumuloDataStoreStatsTest_AccumuloDataStoreStatsTest0_join_name_geom_dtg_v8")
```

Comment:
it would probably be good to create a new schema for just this test, so that other tests don't expect the table and die. also, to insulate from table naming / version changes, you should be able to get the table name like:
```
ds.manager.indices(sft).collectFirst {
  case i: AccumuloJoinIndex if i.attributes.head == "foo" => i.getTableNames().head
}
```

Reply:
I like the ds.manager call!  I agree it'd be good to create a new schema just for this; that said, this test is already relying on the order of the blocks, so I don't feel bad about leaving it as is.  Thoughts?

Reply:
:shrug: